engine/core/src/main/java/org/datacleaner/configuration/RemoteServerConfigurationImpl.java

```diff
@@ -19,20 +19,54 @@
  */
 package org.datacleaner.configuration;
 
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import org.datacleaner.job.concurrent.ScheduledTaskRunner;
+import org.datacleaner.job.concurrent.TaskListener;
+import org.datacleaner.job.concurrent.TaskRunner;
+import org.datacleaner.job.tasks.Task;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Implementation of {@link RemoteServerConfiguration}.
  */
 public class RemoteServerConfigurationImpl implements RemoteServerConfiguration {
 
+    private static final Logger logger = LoggerFactory.getLogger(RemoteServerConfigurationImpl.class);
+    private static final int TEST_CONNECTION_TIMEOUT = 15 * 1000; // [ms]
+    private static final long ERROR_DELAY_MIN = 1;
+    private static final long OK_DELAY_MIN = 5;
+    private final Map<String, RemoteServerState> actualStateMap = Collections.synchronizedMap(new HashMap<>());
+    private ServerStatusTask serverStatusTask;
+    private ScheduledTaskRunner scheduledTaskRunner;
+    private List<RemoteServerStateListener> listeners = Collections.synchronizedList(new ArrayList());
     protected final List<RemoteServerData> remoteServerDataList;
 
-    public RemoteServerConfigurationImpl(List<RemoteServerData> serverData) {
+    public RemoteServerConfigurationImpl(List<RemoteServerData> serverData, TaskRunner taskRunner) {
         remoteServerDataList = new ArrayList<>(serverData);
+        for (RemoteServerData remoteServerData : serverData) {
+            actualStateMap.put(remoteServerData.getServerName(), RemoteServerState.UNKNOWN);
+        }
+
+        if (taskRunner == null || !(taskRunner instanceof ScheduledTaskRunner)) {
+            logger.warn("Task runner isn't ScheduledTaskRunner. Remote server status task won't be scheduled.");
```

Comment:
Maybe log this at info level, because now a warning logged each time I start my vanilla DataCleaner install.